sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -63,29 +82,39 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
     super(nextPolicy, options);
   }
 
-  private parseWWWAuthenticate(www_authenticate: string): string {
+  private parseWWWAuthenticate(www_authenticate: string): {
+    authorization: string,
+    resource: string
+  } {
+    const returnValue = {
+      authorization: "",
+      resource: ""
+    };
     // Parses an authentication message like:
     // ```
     // Bearer authorization="some_authorization", resource="https://some.url"
     // ```
-    let authenticateArray = www_authenticate.split(" ");
-
-    // Remove the "Bearer" piece
-    delete authenticateArray[0];
+    let spaceSep = www_authenticate.split(" ");
 
     // Split the KV comma-separated list
-    let commaSep = authenticateArray.join().split(",");
-    for (let item of commaSep) {
-      // Split the key/value pairs
-      let kv = item.split("=");
-      if (kv[0].trim() == "resource") {
-        // Remove the quotations around the string
-        let resource = kv[1].trim().replace(/['"]+/g, '');
-
-        return resource;
+    for (let spaceItem of spaceSep) {
+      let commaSep = spaceItem.split(",");
+      for (let commaItem of commaSep) {
+        // Split the key/value pairs
+        let kv = commaItem.split("=");
+        if (kv[0].trim() == "authorization") {
```

Comment:
Nit: potentially trimming the same token string twice. More performant if you just do it once and store to a local value.

Reply:
Thanks! I'll push a fix.